.github/workflows/tests.yml

```
@@ -0,0 +1,47 @@
+name: Test
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        go: ["1.18", "1.19", "1.20"]
+        os:
+          [
+            "windows-2019",
+            "windows-2022",
+            "ubuntu-20.04",
+            "ubuntu-22.04",
+            "macos-11",
+            "macos-12",
+            "macos-13",
+          ]
```

Making sure the repository works for every OS.

---

@JoseAngel1196 Lets only check for the latest OS.
```suggestion
          [
            "windows-latest",
            "ubuntu-latest",
            "macos-latest",
          ]
```

---

Done!